src/components/Menu/MenuItem/MenuItem.jsx

```diff
@@ -209,8 +210,13 @@ const MenuItem = forwardRef(
       };
     }, [children, hasOpenSubMenu]);
 
-    const shouldShowTooltip = isTitleHoveredAndOverflowing || disabled;
-    const tooltipContent = disabled ? disableReason : title;
+    const shouldShowTooltip = isTitleHoveredAndOverflowing || disabled || tooltipContent;
+
+    const getTooltipContent = () => {
```

Comment:
no need to put this logic inside a function which will create for each render.
you can set this logic as regular code inside the component.
If the logic were complex I would suggest use memo but it's only few conditions